rotkehlchen/globaldb/handler.py

```diff
@@ -257,14 +258,33 @@ def add_setting_value(name: str, value: Any) -> None:
     def add_asset(
             asset_id: str,
             asset_type: AssetType,
-            data: Union[EvmToken, dict[str, Any]],
+            data: Union[EvmToken, CustomAsset, dict[str, Any]],
     ) -> None:
         """
         Add an asset in the DB. Either an ethereum token or a user asset.
 
         If it's a user asset the data should be typed. As given in by marshmallow.
 
         May raise InputError in case of error, meaning asset exists or some constraint hit"""
+        if asset_type == AssetType.CUSTOM_ASSET:
+            # TODO: change later to a normal (non-hacky) approach
+            asset = cast(CustomAsset, data)
+            try:
+                with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                    write_cursor.execute(
+                        'INSERT INTO assets(identifier, name, type) VALUES(?, ?, ?)',
+                        (asset_id, asset.name, AssetType.CUSTOM_ASSET.serialize_for_db()),
+                    )
+                    write_cursor.execute(
+                        'INSERT INTO custom_assets(identifier, notes, type) VALUES(?, ?, ?)',
+                        (asset_id, asset.notes, asset.custom_asset_type),
+                    )
+            except sqlite3.IntegrityError as e:
+                raise InputError(
+                    f'Failed to add asset {asset_id} into the assets table due to {e!s}',
+                ) from e
+            return
```

Comment:
This already exists at `add_custom_asset`

Reply:
Same reply as above